services/s3/src/it/java/software/amazon/awssdk/services/s3/utils/S3TestUtils.java

```diff
@@ -29,6 +34,56 @@
 import software.amazon.awssdk.testutils.Waiter;
 
 public class S3TestUtils {
+    private static final String TEST_BUCKET_PREFIX = "s3-test-bucket-";
+    private static final String NON_DNS_COMPATIBLE_TEST_BUCKET_PREFIX = "s3.test.bucket.";
+
+    public static String getTestBucket(S3Client s3) {
+        return getBucketWithPrefix(s3, TEST_BUCKET_PREFIX);
+    }
+
+    public static String getNonDnsCompatibleTestBucket(S3Client s3) {
+        return getBucketWithPrefix(s3, NON_DNS_COMPATIBLE_TEST_BUCKET_PREFIX);
+    }
+
+    private static String getBucketWithPrefix(S3Client s3, String bucketPrefix) {
+        String testBucket =
+            s3.listBuckets()
+              .buckets()
+              .stream()
+              .map(Bucket::name)
+              .filter(name -> name.startsWith(bucketPrefix))
+              .findAny()
+              .orElse(null);
+
+        if (testBucket == null) {
+            String newTestBucket = bucketPrefix + UUID.randomUUID();
```

Comment:
The bucket name needs to be unique to prevent different accounts from interfering with each other. Random seemed like the best method.

Reply:
Usually I pass this in as an env variable, so it's deterministic based on stack, but I'm happy to treat this feedback as FYI/did-you-consider rather than 'we must change this'.

Reply:
I wouldn't even want it to be deterministic based on stack. If the bucket gets deleted by hand (which happens frequently when cleaning other stuff up), then this test would fail until the bucket finishes being deleted (which can take multiple hours). We've been bitten by that exact scenario before.

Reply:
Ok we can leave it as-is for the sake of getting this out, but a couple of points : 1) bucket deletion should not take multiple hours because we're keeping our keys clean!  2) in the unlikely event the bucket isn't actually there we can make creating it part of the initialization flow. The only thing we'd be doing differently is using a deterministic name. Ideally most of the time we wouldn't need to create nor delete the bucket.